packages/analytics/amplify_analytics_pinpoint_dart/lib/src/impl/analytics_client/endpoint_client/endpoint_id_manager.dart

```diff
@@ -0,0 +1,79 @@
+import 'package:amplify_analytics_pinpoint_dart/src/impl/analytics_client/endpoint_client/endpoint_store_keys.dart';
+import 'package:amplify_analytics_pinpoint_dart/src/impl/flutter_provider_interfaces/legacy_native_data_provider.dart';
+import 'package:amplify_core/amplify_core.dart';
+import 'package:amplify_secure_storage_dart/amplify_secure_storage_dart.dart';
+
+/// {@template amplify_analytics_pinpoint_dart.endpoint_info_store}
+/// Manages retrieval, storage, and management of Pinpoint Endpoint id
+/// {@endtemplate}
+class EndpointIdManager {
+  /// {@macro amplify_analytics_pinpoint_dart.endpoint_info_store}
+  EndpointIdManager({
+    required SecureStorageInterface store,
+    LegacyNativeDataProvider? legacyNativeDataProvider,
+    String? pinpointAppId,
+  })  : _store = store,
+        _legacyNativeDataProvider = legacyNativeDataProvider,
+        _pinpointAppId = pinpointAppId;
+
+  final SecureStorageInterface _store;
+  final LegacyNativeDataProvider? _legacyNativeDataProvider;
+  final String? _pinpointAppId;
```

Comment:
Why nullable? How are endpoint IDs segregated if not by pinpoint app ID?

Reply:
> Analytics and Auth will share the same endpoint by using the same SecureStorageInterface scope

How will that work? Can you add unit tests which showcase this? What if the Analytics category and Auth category point to different Pinpoint App IDs? And what if Push points to yet another? Why would Analytics/Auth share a scope in this case? And which one owns the `SecureStorageInterface`?

Also, still not sure why this is null here.

Reply:
You have a good point here ... I was copying iOS behavior here but it is very strange.  I'm going to double check this behavior with Abdallah. 

In the meantime, here's more context: 

Analytics and Auth will use the same endpoint because they use the same SecureStorage database to retrieve and store endpoints.  That will be more obvious in the next PR to refactor endpoint information management. 

I will add unit tests - not sure about testing if Analytics/Auth share the same endpoint.  The idea is that Analytics and Auth will use the same `FlutterEndpointInformationManager` class so by using the same implementation it should be guaranteed the data used is the same. 


The expected behavior is NOT for Analytics and Auth to use different Pinpoint App Ids.  In general it's not expected that any of the Analytics, Auth, Notification category Pinpoint App Ids should be different at all per our recent Analytics group discussions.  Again, I'll double check that with Abdallah. 

Analytics/Auth share a scope because it was assumed they must use the same Pinpoint App Id.  

They both own the storage.  The idea is that even if Analytics is never added, Auth can create and manage endpoint itself and vice versa. 

The PinpointAppId is nullable here because it reflects the philosophy that the Endpoint Id is the same between Analytics and Auth.  It's only needed for retrieving legacy Endpoint Id from Android (and NOT iOS).

Reply:
In this PR I refactor Endpoint logic: https://github.com/aws-amplify/amplify-flutter/pull/2639

Since the behavior in this PR is not different than previous implementations, and the next PR is related to Endpoint information storage, I recommend we tackle the refactor of the logic in that followup PR. 

I also plan to add tests there as well.

Reply:
> not sure about testing if Analytics/Auth share the same endpoint

That's not needed for this PR. But it should be possible to have some unit tests which shows how two different `EndpointIdManager`'s with different pinpoint app IDs would interact since it's not just endpoint IDs which are being stored but also global properties etc.

Reply:
After speaking with native teams and Chris, I've decided the best approach is to implement Android's strategy of using the pinpointAppId to store the endpoint information (iOS just uses a hardcoded key which is what we do now). 

I'll do that in the dedicated endpoint refactor PR and add unit tests there.  Made a note of this to make sure I do that.